src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -247,8 +248,8 @@ impl SyscallHandler for BusinessLogicSyscallHandler {
     ) -> Result<(), SyscallHandlerError> {
         let response_data = self._call_contract(syscall_name, vm, syscall_ptr.clone())?;
         // TODO: Should we build a response struct to pass to _write_syscall_response?
-        self._write_syscall_response(response_data, vm, syscall_ptr);
-        Ok(())
+        // self._write_syscall_response(response_data, vm, syscall_ptr);
+        todo!()
```

Comment:
what is left for removing this todo!() ?

Reply:
I add the todo tag here to make clear that the `_call_contract_and_write_response` method is unfinished
[Python implementation](https://github.com/starkware-libs/cairo-lang/blob/de741b92657f245a50caab99cfaef093152fd8be/src/starkware/starknet/core/os/syscall_utils.py#L469)

First, we need to finish  the `_call_contract` method and then we can call `_write_syscall_response`

Reply:
Ohh I see